eth/executionclient/observability.go

```diff
@@ -0,0 +1,98 @@
+
+const (
+	observabilityComponentName      = "github.com/ssvlabs/ssv/eth/executionclient"
+	observabilityComponentNamespace = "ssv.el"
+)
+
+type executionClientStatus string
+
+const (
+	statusSyncing executionClientStatus = "syncing"
+	statusFailure executionClientStatus = "failure"
+	statusReady   executionClientStatus = "ready"
+)
+
+var (
+	meter = otel.Meter(observabilityComponentName)
+
+	latencyHistogram = observability.GetMetric(
+		fmt.Sprintf("%s.latency.duration", observabilityComponentNamespace),
+		func(metricName string) (metric.Float64Histogram, error) {
+			return meter.Float64Histogram(
+				metricName,
+				metric.WithUnit("s"),
+				metric.WithDescription("execution client latency in seconds"),
+				metric.WithExplicitBucketBoundaries(observability.SecondsHistogramBuckets...),
+			)
+		},
+	)
+
+	syncingDistanceGauge = observability.GetMetric(
+		fmt.Sprintf("%s.syncing.distance", observabilityComponentNamespace),
+		func(metricName string) (metric.Int64Gauge, error) {
+			return meter.Int64Gauge(
+				metricName,
+				metric.WithUnit("{block}"),
+				metric.WithDescription("execution client syncing distance which is a delta between highest and current blocks"))
+		},
+	)
+
+	clientStatusGauge = observability.GetMetric(
+		fmt.Sprintf("%s.syncing.status", observabilityComponentNamespace),
+		func(metricName string) (metric.Int64Gauge, error) {
+			return meter.Int64Gauge(
+				metricName,
+				metric.WithDescription("execution client syncing status"))
+		},
+	)
+
+	lastProcessedBlockGauge = observability.GetMetric(
+		fmt.Sprintf("%s.syncing.last_processed_block", observabilityComponentNamespace),
+		func(metricName string) (metric.Int64Gauge, error) {
+			return meter.Int64Gauge(
+				metricName,
+				metric.WithUnit("{block_number}"),
+				metric.WithDescription("last processed block by execution client"))
+		},
+	)
+)
+
+func executionClientAddrAttribute(value string) attribute.KeyValue {
+	eventNameAttrName := fmt.Sprintf("%s.addr", observabilityComponentNamespace)
+	return attribute.String(eventNameAttrName, value)
+}
+
+func executionClientStatusAttribute(value executionClientStatus) attribute.KeyValue {
+	eventNameAttrName := fmt.Sprintf("%s.status", observabilityComponentNamespace)
+	return attribute.String(eventNameAttrName, string(value))
+}
+
+func recordExecutionClientStatus(ctx context.Context, status executionClientStatus, nodeAddr string) {
+	resetExecutionClientStatusGauge(ctx, nodeAddr)
+
+	clientStatusGauge.Record(ctx, 1,
+		metric.WithAttributes(executionClientAddrAttribute(nodeAddr)),
+		metric.WithAttributes(executionClientStatusAttribute(status)),
+	)
+}
+
+func resetExecutionClientStatusGauge(ctx context.Context, nodeAddr string) {
+	for _, status := range []executionClientStatus{statusReady, statusSyncing, statusFailure} {
+		clientStatusGauge.Record(ctx, 0,
+			metric.WithAttributes(executionClientAddrAttribute(nodeAddr)),
+			metric.WithAttributes(executionClientStatusAttribute(status)),
+		)
+	}
+}
```

Comment:
can just be one function with a `status bool` parameter IMO, to reduce boilerplate

Reply:
do you mean merge these two functions and remove `nodeAddr` attribute from the metric?

Reply:
oh sorry, i guess i didnt understand the purpose of the reset

is it really how to express enums with otel?

can't we instead just encode the status in the value? (so 0 = "syncing", 1 = "failure", etc..)

Reply:
This isn’t specifically OTeL-specific. It is the same pattern in Prometheus. The problem definition is: “In the metric, I want to indicate a value that isn’t a number.” Observability frameworks use number as _metric values_. One option is to do what you suggested - use indicative enums, another option is to represent it as a label with a boolean value (1/0). 
Both options have pros and cons. I cannot 100% say I fully evaluated both. We had discussion about this with @zktaiga . 
What we know for sure - the problem remains and we need to be consistent with the pattern. Probably less important which option we pick, more important to stick with the same pattern everywhere.